internal/cloud/run.go

```diff
@@ -502,6 +505,7 @@ func getDesiredRunStatus(run *tfe.Run, policyChecksEnabled bool, costEstimateEna
 	desiredStatus := []tfe.RunStatus{
 		tfe.RunPolicySoftFailed,
 		tfe.RunPlannedAndFinished,
+		tfe.RunPlannedAndSaved,
 		tfe.RunApplied,
 	}
```

Comment:
Not that this would be an issue but I think it's better to only include this in the desired run status if SavePlan evaluates to true.
```suggestion
	desiredStatus := []tfe.RunStatus{
		tfe.RunPolicySoftFailed,
		tfe.RunPlannedAndFinished,
		tfe.RunApplied,
	}
	
	if run.SavePlan {
		desiredStatus = append(desiredStatus, tfe.RunPlannedAndSaved)
	}
```

Reply:
Thank you for your suggestion. [8ab2990](https://github.com/hashicorp/tfc-workflows-tooling/pull/57/commits/8ab299045313b8c187d650be6c9d239f321685b3)